src/components/Comments.jsx

```diff
@@ -1,16 +1,49 @@
 import { useContext, useEffect, useState } from "react";
 import { UserContext, UserProvider } from "../contexts/UserContext";
-import { getAllArticles, getCommentsByArticleID } from "../utils/api";
+import {
+  deleteCommentbyID,
+  getAllArticles,
+  getCommentsByArticleID,
+} from "../utils/api";
+import popSound from "../assets/popSound.mp3";
 
 import { useParams } from "react-router-dom";
+import AddNewComment from "./AddNewComment";
 
-function Comments() {
+function Comments({ showComments, setShowComments }) {
   const [comments, setComments] = useState([]);
-
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(false);
   const { user, setUser } = useContext(UserContext);
   const { article_id } = useParams();
+  const [addComment, setAddComment] = useState(false);
+  //const [isDeleted, setIsDeleted] = useState(false)
+
+  function refreshComments() {
+    setShowComments(!showComments);
+    new Audio(popSound).play();
+  }
+
+  function popwithAddComment() {
+    setAddComment(true);
+    new Audio(popSound).play();
+  }
+
+  function popwithdeleteComment(comment_id) {
+    deleteCommentbyID(comment_id);
```

Comment:
If this API request fails (for example due to bad internet), the user needs to be informed either visually or via some feedback that the comment has not successfully been deleted. Currently it still is deleted in state even if the API request fails.